interop/shared/src/main/scala/scalaz/zio/interop/future.scala

```diff
@@ -0,0 +1,43 @@
+package scalaz.zio
+package interop
+
+import scala.concurrent.Future
+
+object future {
```

Comment:
Hey @jdegoes I could you a bit of early feedback on the API while I work on the impl + tests